idp/openid: fix registration step.

mhilton opened the pull request:
If an OpenID Connect login requires the user to register then the
login attempt would fail. The login process now makes sure that all
"candid-login" cookies are created with the same path. Also the
registration template ensures the state parameter is echoed back.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/33/

rogpeppe commented on docs/configuration.md:
```diff
@@ -226,6 +226,115 @@ will be used.
+Candid will prompt for a username and password and attempt to use those
+to authenticate with the LDAP server.
+
+`name` is the name to use for the LDAP IDP instance, it is possible
```
s/, it/. It/

rogpeppe commented on docs/configuration.md:
```diff
@@ -226,6 +226,115 @@ will be used.
+
+`name` is the name to use for the LDAP IDP instance, it is possible
+to configure more than one LDAP IDP on a given candid server and this
+allows them to be identified, name will be used in the login URL.
```
s/, name will /. The name will/

rogpeppe commented on docs/configuration.md:
````diff
@@ -226,6 +226,115 @@ will be used.
+  client-secret: 8aENrwCL/+PU87ROkXwMB+09xe0=
+```
+
+The Google identity provider uses OpenID connect to log in using Google
````
s/connect/Connect/ ?

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/34/

albertodonato commented on docs/configuration.md:
```diff
@@ -226,6 +226,115 @@ will be used.
+
+`group-query-filter` contains the filter candid uses when finding group
+memberships for a user.  The filter is specified as a template (see
+https://golang.org/pkg/text/template) where the value of .User will be
```
maybe backtick-quote `.User` too ?

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/35/

mhilton commented:
:shipit:

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/36/

jujugui commented:
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

mhilton committed:
idp/openid: fix registration step.

If an OpenID Connect login requires the user to register then the
login attempt would fail. The login process now makes sure that all
"candid-login" cookies are created with the same path. Also the
registration template ensures the state parameter is echoed back.
